core/storage/pager.rs

```diff
@@ -867,6 +925,12 @@ impl Pager {
         _alloc_mode: BtreePageAllocMode,
     ) -> Result<IOResult<BTreePage>> {
         let page = return_if_io!(self.allocate_page());
+        if let Some(wal) = &self.wal {
+            let max_frame = wal.borrow().get_max_frame_in_wal();
+            let seq = wal.borrow().checkpoint_seq();
+            // brand new page gets share max frame + 1
+            page.set_wal_tag(max_frame + 1, seq);
+        }
```

Comment:
also, do we keep all pages of a txn in memory? do we not drop them from the memory and reload again later?